olp-cpp-sdk-authentication/src/AuthorizeRequest.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "olp/authentication/AuthorizeRequest.h"
+
+#include <sstream>
+
+namespace olp {
+namespace authentication {
+
+std::string AuthorizeRequest::CreateKey() const {
+  std::stringstream out;
+  out << service_id_;
+  if (GetContractId()) {
+    out << "[" << GetContractId().get() << "]";
+  }
+  out << "{";
+  for (const auto& actions : actions_) {
+    out << " (" << actions.first;
+    if (actions.second) {
+      out << ", " << actions.second.get();
+    }
+    out << ")";
+  }
+  out << "}";
+  out << "^"
+        << (GetOperatorType() == DecisionApiOperatorType::kAnd ? "AND"
+                                                              : "OR");
+  return out.str();
```

Comment:
This is too much. The sole purpose of the key is to have nice print log. I would say service_id and contract_id are totally sufficient.